### PR TITLE
SIMSBIOHUB-84

### PR DESCRIPTION
**Pull request by cgarrettjones**
# Overview

- added concept of custome security reason specific to a dataset or artifact
- collapsed existing migrations into the model

## Links to Jira tickets

- [SIMSBIOHUB-84](https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-84)

## Description of relevant changes

- {List all relevant changes, in particular anything that will help the reviewers test/verify this PR}

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

File: database/src/migrations/release.0.8.0/biohub.sql [hunk 1]
```diff
@@ -968,19 +995,19 @@ COMMENT ON TABLE submission_observation IS 'A listing of historical data submiss
...
+    spatial_component                  jsonb             NOT NULL,
+    geometry                           point,
+    geography                          point,
     secured_spatial_component          jsonb,
-    secured_geometry                   geometry(geometry, 3005),
-    secured_geography                  geography(geometry),
-    create_date                        timestamptz(6)              DEFAULT now() NOT NULL,
-    create_user                        integer                     NOT NULL,
+    secured_geometry                   point,
+    secured_geography                  point,
```
Comment (NickPhura) [inline database/src/migrations/release.0.8.0/biohub.sql]: Does the point type work the same with the PostGIS functions? Doing spatial intersect queries, etc?

Comment (NickPhura) [reply database/src/migrations/release.0.8.0/biohub.sql]: For example: `spatial-repository.ts -> _whereBoundaryIntersects`.

Comment (cgarrettjones) [reply database/src/migrations/release.0.8.0/biohub.sql]: No, of course not! having logical to physical mapping issues with the tool. Nice catch! please find corrected.

**Review by al-rosenthal — APPROVED** [review]
🥡

**Review by NickPhura — APPROVED** [review]

**Pull request merged**